lib/models/search/button_result.dart

```diff
@@ -0,0 +1,7 @@
+import "package:photos/ui/components/button_widget.dart";
+
+class ButtonResult {
+  final ButtonAction? action;
+  final Exception? exception;
+  ButtonResult({required this.action, this.exception});
```

Comment:
Can reduce verbosity by not using named parameters? Both params are of different types.